### PR TITLE
Add Evaluate Selection option to Script Editor

### LINKED ISSUE #24074: Evaluate expression inside script editor
Basically this (from VS Code):
screen_recorder_plus_video_2018_30_11_01_01_54

Evaluate an expression with keyboard shortcut (something that's somewhat already in inspector's number input boxes) Allows for quick maths, so you don't need to leave some weird expressions in the code like `node.position += 131 + 34` or fire up a calculator each time you need this.

### PR DESCRIPTION
**Pull request by KoBeWi**
Closes #24074

Evaluate
(the last one is a joke, nothing happens if there is an error)

### CODE REVIEW

**Comment by akien-mga** [comment]
Does it work with UndoRedo?

**Comment by KoBeWi** [comment]
> Does it work with UndoRedo?

Yes, it uses `insert_text_at_cursor()` in the end.

**Comment by Xrayez** [comment]
Does it support evaluating multiple lines at once? 
Such a good replacement for Python shell to calculate stuff!

**Comment by KoBeWi** [comment]
> Does it support evaluating multiple lines at once?

No, it only evaluates first line when multiple are selected. Maybe I could do that with `split()`...

File: editor/plugins/script_text_editor.cpp [hunk 1]
```diff
@@ -1147,6 +1148,32 @@ void ScriptTextEditor::_edit_option(int p_op) {
...
+						results.append(whitespace + (String)result);
+					} else {
+						results.append(line);
+					}
+				} else {
+					results.append(line);
+				}
+			}
+
+			code_editor->get_text_edit()->begin_complex_operation(); //prevents creating a two-step undo
```
Comment (bojidar-bg) [inline editor/plugins/script_text_editor.cpp]: Will it actually create a two-step undo though?

Comment (KoBeWi) [reply editor/plugins/script_text_editor.cpp]: Yes, I added the fix because it was happening.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**